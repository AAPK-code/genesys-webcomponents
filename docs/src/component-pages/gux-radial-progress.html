<html>
  <head>
    <title>gux-radial-progress</title>
  </head>
  <body>
    <script src="${CDN_URL}index.js"></script>
    <script>
      webcomponentsDocsMain(`
<gux-radial-progress></gux-radial-progress>

<gux-radial-progress
    value=0
    max=100
></gux-radial-progress>

<gux-radial-progress
    value=3
    max=10
></gux-radial-progress>

<gux-radial-progress
    value=10
    max=10
></gux-radial-progress>
`);
    </script>
  </body>
</html>
