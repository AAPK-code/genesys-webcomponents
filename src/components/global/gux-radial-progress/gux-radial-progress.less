// Variables part

// Style
gux-radial-progress {
  > div {
    display: block;

    .svgContainer {
      display: block;
    }

    .dynamicCircle {
      stroke-dasharray: 163.28;
      transform: rotate(-90deg);
      transform-origin: 50% 50%;
      fill: none;
      stroke: #75A8FF;
      stroke-width: 5;
    }
    .staticCircle {
      fill: none;
      stroke: #deeaff;
      stroke-width: 5;
    }
    .percentage {
      text-anchor: middle;
      dominant-baseline: central;
      fill: #444A52;
      font-size: 12px;
    }
  }
}

// Theming

// Dark
.gux-radial-progress-dark-theme {
    color: @gux-off-white;
    > div {
      .percentage {
        fill: #fdfdfd;
      }
    }
};

.gux-dark-theme {
    gux-radial-progress {
        .gux-radial-progress-dark-theme();
    }
}

gux-radial-progress.gux-dark-theme {
    .gux-radial-progress-dark-theme();
}

// Light
.gux-radial-progress-light-theme {
    color: @gux-grey-2;
};

.gux-light-theme {
    gux-radial-progress {
        .gux-radial-progress-light-theme();
    }
}

gux-radial-progress.gux-light-theme {
    .gux-radial-progress-light-theme();
}

// Default Theme
gux-radial-progress {
    .gux-radial-progress-light-theme();
}