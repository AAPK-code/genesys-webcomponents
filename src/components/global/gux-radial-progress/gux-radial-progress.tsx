import { Component, Prop } from '@stencil/core';

@Component({
  styleUrl: 'gux-radial-progress.less',
  tag: 'gux-radial-progress'
})
export class GuxRadialProgress {
  /**
   * Indicates the percentage displayed in the progress spinner
   */
  @Prop()
  percentage: number;

  render() {
    return (
      <div>
        <svg class="svgContainer" width="65px" height="65px" viewBox="0 0 65 65">
          <filter id="dropshadow">
            <feGaussianBlur in="SourceGraphic" stdDeviation="2"/>
            <feOffset dx="0" dy="0" result="offsetblur"/>
            <feMerge>
              <feMergeNode />
              <feMergeNode in="SourceGraphic"/>
            </feMerge>
          </filter>
          <circle
            cx="50%"
            cy="50%"
            r="26"
            class="staticCircle"
          />
          <circle
            cx="50%"
            cy="50%"
            r="26"
            class="dynamicCircle"
            stroke-dashoffset={163.28 * (1 - (this.percentage / 100))}
            filter="url(#dropshadow)"
          />

          <text
            x="50%"
            y="50%"
            class="percentage"
          >
            {`${this.percentage}%`}
          </text>
        </svg>
      </div>
    );
  }
}
