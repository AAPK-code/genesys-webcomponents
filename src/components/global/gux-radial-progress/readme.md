# gux-radial-progress

Use when the progress of a task is known, this component displays a radial indicator and the completed percentage of 
the task.

<!-- Auto Generated Below -->


## Properties

| Property     | Attribute    | Description                                                | Type     |
| ------------ | ------------ | ---------------------------------------------------------- | -------- |
| `percentage` | `percentage` | Indicates the percentage displayed in the progress spinner | `number` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
