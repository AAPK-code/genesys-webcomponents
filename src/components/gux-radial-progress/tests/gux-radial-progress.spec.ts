import { newSpecPage } from '@stencil/core/testing';
import { GuxRadialProgress } from '../gux-radial-progress';

describe('gux-radial-progress', () => {
  describe('#render', () => {
    [
      '<gux-radial-progress></gux-radial-progress>',
      '<gux-radial-progress value="0"></gux-radial-progress>',
      '<gux-radial-progress value="10"></gux-radial-progress>',
      '<gux-radial-progress value="100"></gux-radial-progress>',
      '<gux-radial-progress value="123"></gux-radial-progress>',
      '<gux-radial-progress value="200"></gux-radial-progress>',
      '<gux-radial-progress value="10" max="100"></gux-radial-progress>',
      '<gux-radial-progress value="10" max="10"></gux-radial-progress>',
      '<gux-radial-progress value="test" max="100"></gux-radial-progress>',
      '<gux-radial-progress value="-123" max="100"></gux-radial-progress>',
      '<gux-radial-progress value="200" max="100"></gux-radial-progress>',
      '<gux-radial-progress value="10" max="test"></gux-radial-progress>'
    ].forEach((input, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({
          components: [GuxRadialProgress],
          html: input
        });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
