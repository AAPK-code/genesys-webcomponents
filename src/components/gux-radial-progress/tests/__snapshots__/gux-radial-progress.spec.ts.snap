// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-radial-progress #render should render component as expected (1) 1`] = `
<gux-radial-progress>
  <div aria-busy="true">
    <div class="spinner"></div>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (2) 1`] = `
<gux-radial-progress value="0">
  <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="0" role="progressbar">
    <svg class="svg-container" height="65px" viewBox="0 0 65 65" width="65px">
      <filter id="dropshadow-1">
        <feGaussianBlur in="SourceGraphic" stdDeviation="2"></feGaussianBlur>
        <feOffset dx="0" dy="0" result="offsetblur"></feOffset>
        <feMerge>
          <feMergeNode></feMergeNode>
          <feMergeNode in="SourceGraphic"></feMergeNode>
        </feMerge>
      </filter>
      <circle class="static-circle" cx="50%" cy="50%" r="26"></circle>
      <circle class="dynamic-circle" cx="50%" cy="50%" filter="url(#dropshadow-1)" r="26" stroke-dasharray="163.36281798666926" stroke-dashoffset="163.36281798666926"></circle>
      <text class="percentage" dominant-baseline="central" x="50%" y="50%">
        0%
      </text>
    </svg>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (3) 1`] = `
<gux-radial-progress value="10">
  <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="10" role="progressbar">
    <svg class="svg-container" height="65px" viewBox="0 0 65 65" width="65px">
      <filter id="dropshadow-2">
        <feGaussianBlur in="SourceGraphic" stdDeviation="2"></feGaussianBlur>
        <feOffset dx="0" dy="0" result="offsetblur"></feOffset>
        <feMerge>
          <feMergeNode></feMergeNode>
          <feMergeNode in="SourceGraphic"></feMergeNode>
        </feMerge>
      </filter>
      <circle class="static-circle" cx="50%" cy="50%" r="26"></circle>
      <circle class="dynamic-circle" cx="50%" cy="50%" filter="url(#dropshadow-2)" r="26" stroke-dasharray="163.36281798666926" stroke-dashoffset="147.02653618800232"></circle>
      <text class="percentage" dominant-baseline="central" x="50%" y="50%">
        10%
      </text>
    </svg>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (4) 1`] = `
<gux-radial-progress value="100">
  <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="100" role="progressbar">
    <svg class="svg-container" height="65px" viewBox="0 0 65 65" width="65px">
      <filter id="dropshadow-3">
        <feGaussianBlur in="SourceGraphic" stdDeviation="2"></feGaussianBlur>
        <feOffset dx="0" dy="0" result="offsetblur"></feOffset>
        <feMerge>
          <feMergeNode></feMergeNode>
          <feMergeNode in="SourceGraphic"></feMergeNode>
        </feMerge>
      </filter>
      <circle class="static-circle" cx="50%" cy="50%" r="26"></circle>
      <circle class="dynamic-circle" cx="50%" cy="50%" filter="url(#dropshadow-3)" r="26" stroke-dasharray="163.36281798666926" stroke-dashoffset="0"></circle>
      <text class="percentage" dominant-baseline="central" x="50%" y="50%">
        100%
      </text>
    </svg>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (5) 1`] = `
<gux-radial-progress value="123">
  <div aria-busy="true">
    <div class="spinner"></div>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (6) 1`] = `
<gux-radial-progress value="200">
  <div aria-busy="true">
    <div class="spinner"></div>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (7) 1`] = `
<gux-radial-progress max="100" value="10">
  <div aria-valuemax="100" aria-valuemin="0" aria-valuenow="10" role="progressbar">
    <svg class="svg-container" height="65px" viewBox="0 0 65 65" width="65px">
      <filter id="dropshadow-6">
        <feGaussianBlur in="SourceGraphic" stdDeviation="2"></feGaussianBlur>
        <feOffset dx="0" dy="0" result="offsetblur"></feOffset>
        <feMerge>
          <feMergeNode></feMergeNode>
          <feMergeNode in="SourceGraphic"></feMergeNode>
        </feMerge>
      </filter>
      <circle class="static-circle" cx="50%" cy="50%" r="26"></circle>
      <circle class="dynamic-circle" cx="50%" cy="50%" filter="url(#dropshadow-6)" r="26" stroke-dasharray="163.36281798666926" stroke-dashoffset="147.02653618800232"></circle>
      <text class="percentage" dominant-baseline="central" x="50%" y="50%">
        10%
      </text>
    </svg>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (8) 1`] = `
<gux-radial-progress max="10" value="10">
  <div aria-valuemax="10" aria-valuemin="0" aria-valuenow="10" role="progressbar">
    <svg class="svg-container" height="65px" viewBox="0 0 65 65" width="65px">
      <filter id="dropshadow-7">
        <feGaussianBlur in="SourceGraphic" stdDeviation="2"></feGaussianBlur>
        <feOffset dx="0" dy="0" result="offsetblur"></feOffset>
        <feMerge>
          <feMergeNode></feMergeNode>
          <feMergeNode in="SourceGraphic"></feMergeNode>
        </feMerge>
      </filter>
      <circle class="static-circle" cx="50%" cy="50%" r="26"></circle>
      <circle class="dynamic-circle" cx="50%" cy="50%" filter="url(#dropshadow-7)" r="26" stroke-dasharray="163.36281798666926" stroke-dashoffset="0"></circle>
      <text class="percentage" dominant-baseline="central" x="50%" y="50%">
        100%
      </text>
    </svg>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (9) 1`] = `
<gux-radial-progress max="100" value="test">
  <div aria-busy="true">
    <div class="spinner"></div>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (10) 1`] = `
<gux-radial-progress max="100" value="-123">
  <div aria-busy="true">
    <div class="spinner"></div>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (11) 1`] = `
<gux-radial-progress max="100" value="200">
  <div aria-busy="true">
    <div class="spinner"></div>
  </div>
</gux-radial-progress>
`;

exports[`gux-radial-progress #render should render component as expected (12) 1`] = `
<gux-radial-progress max="test" value="10">
  <div aria-busy="true">
    <div class="spinner"></div>
  </div>
</gux-radial-progress>
`;
